### DISCUSSION TITLE
weasis Import DICOM via Query Retrieve (DIMSE C-FIND) is not working

### DISCUSSION THREAD

**Discussion by sig-wi** [nroduit/Weasis#518] [Q&A]
Import DICOM via Query Retrieve does not work with our DICOM Archive. 
The C-FIND request identifier created by this funtion in my understanding violates the DICOM Standard.
It uses STUDY as Query Retrieve Level but asks for (0080,0080) Institution Name, which is a SERIES Level Attribute - see output from Logfile: 
21.02.2024 11:48:49.178 *DEBUG* [DICOM C-FIND] org.dcm4che3.net.Dimse: WEASIS->QFMSCP.TEST0001(1) << 1:C-FIND-RQ Dataset:
(0008,0020) DA [] StudyDate
(0008,0030) TM [] StudyTime
(0008,0050) SH [] AccessionNumber
(0008,0052) CS [STUDY] QueryRetrieveLevel
(0008,0061) CS [] ModalitiesInStudy
(0008,0080) LO [] InstitutionName
(0008,0090) PN [] ReferringPhysicianName
(0008,1030) LO [] StudyDescription
(0010,0010) PN [] PatientName
(0010,0020) LO [] PatientID
(0010,0021) LO [] IssuerOfPatientID
(0010,0030) DA [] PatientBirthDate
(0010,0040) CS [] PatientSex
(0020,000D) UI [] StudyInstanceUID
(0020,0010) SH [] StudyID
(0020,1206) IS [] NumberOfStudyRelatedSeries
(0020,1208) IS [] NumberOfStudyRelatedInstances

Is it possible to change this via config / preferences to make this work with our archive?

**Comment by nroduit** [comment]
We will only change this configuration if it does not comply with the standard.

Searching around, I can't make up my mind about this question, because I can find both statements.

Referring to the current standard, [the field is not explicitly included in the optional fields](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.6.html), neither for the study level nor for the Series level. "Institution name" (0008,0080) can be assigned to the Equipment module (SERIES) or to the General Study module.

If you can show me that it's only in the Series level, we'll modify it. Otherwise, I think your PACS should tolerate it, as this is the first time this limitation has been reported to me.

**Reply by sig-wi** [reply]
Thank you for your answer! Let me outline the rationale (sorry for being so verbose).

PS3.4 [C.3.1.](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.3.html#sect_C.3.1) and [C.3.2](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.3.2.html) define the Query Retrieve Information Models and the associated hierarchy levels.

STUDY Level: 
"_The study level is below the patient level and contains Attributes associated with the Study IE of the Composite IODs as defined in​ PS3.3. A study belongs to a single patient. A single patient may have multiple studies. Study IEs are modality independent._"

PS3.3 [C.7.2](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.2.html) defines the Common Study IE as consisting of General Study Module, Patient Study Module, and Clinical Trial Study Module.
None of these contain (0008,0080) Institution Name **at root level**. It only comes as an attribute within the diverse physician identification sequences (via the Person Identification Macro Attributes). In this sequence (0008,0080) Institution Name expresses the institution a physician works for - either the Referring Physician, one or more Consulting Physicians, Physicians of Record, ... 

PS3.4 [Table C.6-2](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.6.html#table_C.6-2) defines the Study Level Keys for the Patient Root Query/Retrieve Information Model.
PS3.4 [Table C.6-5](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.6.2.html#table_C.6-5) defines the Study Level Keys for the Study Root Query/Retrieve Information Model.
Both tables do not contain (0008,0080) Institution Name but a table row "_All other Attributes at Study Level_" - and this includes all attributes defined in the Study IE.

SERIES Level: 
"_The series level is below the study level and contains Attributes associated with the Series, Frame of Reference and Equipment IEs​ of the Composite IODs as defined in PS3.3. A series belongs to a single study. A single study may have multiple series. Series IEs​ are modality dependent. To accommodate this modality dependence, the set of Optional Keys at the series level includes all Attributes​ defined at the series level from any Composite IOD defined in PS3.3_" 

PS3.4 [Table C.6-3](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.6.html#table_C.6-3) defines the Series Level Attributes for the Patient Root Query/Retrieve Information Model.
The same table is valid for the Study Root Query/Retrieve Information Model.
This table again includes "_All Other Attributes at Series Level_" - which again includes all attributes in the IEs defined for Series Level.

Institution Name is not contained in the General Series nor Enhanced Series Module but in the **General Equipment Module** which belongs the Common Equipment IE. This Institution Name is defined as "_Institution where the equipment that produced the Composite Instances is​ located._" 

In my understanding this is the property that one wants to query.
And this can be different for different series of a study.

So what should a Query/Retrieve SCP return if an SCU requests Institution Name at STUDY Level ???

**Comment by nroduit** [comment]
As it seems not really used, I replaced Institution Name with Study Instance UID
